### PR TITLE
Refactor runAsNonRoot to have named errors and better error messages

### PR DESCRIPTION
**Pull request by klautcomputing**
Same as #27 and #26

### CODE REVIEW

File: cmd/runAsNonRoot.go [hunk 1]
```diff
@@ -8,23 +8,36 @@ import (
...
+	if container.SecurityContext.RunAsNonRoot == nil {
+		result.err = ErrorRunAsNonRootNIL
+		return
+	}
+	if !*container.SecurityContext.RunAsNonRoot {
+		result.err = ErrorRunAsNonRootFalse
+		return
 	}
-
 	return
```
Comment (csgregorian) [inline cmd/runAsNonRoot.go]: ~return~

File: cmd/image.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func checkImage(container apiv1.Container, image imgFlags, result *Result) {
 	}
 
 	if contImage.name == image.name && contImage.tag != image.tag {
-		result.err = 2
```
Comment (jinankjain) [inline cmd/image.go]: This should not be here!

Comment (klautcomputing) [reply cmd/image.go]: You mean in this PR? You are right it should have been in #26 ...

Comment (jinankjain) [reply cmd/image.go]: yeah or it should be moved to different commit

**Pull request merged**